odoo/addons/base/models/avatar_mixin.py

```diff
@@ -25,11 +25,11 @@ class AvatarMixin(models.AbstractModel):
     _avatar_name_field = "name"
 
     # all image fields are base64 encoded and PIL-supported
-    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920")
-    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024")
-    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512")
-    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256")
-    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128")
+    avatar_1920 = fields.Image("Avatar", compute="_compute_avatar_1920")
+    avatar_1024 = fields.Image("Avatar 1024", compute="_compute_avatar_1024")
+    avatar_512 = fields.Image("Avatar 512", compute="_compute_avatar_512")
+    avatar_256 = fields.Image("Avatar 256", compute="_compute_avatar_256")
+    avatar_128 = fields.Image("Avatar 128", compute="_compute_avatar_128")
```

Comment:
This is wrong in some cases, it is now basically relying on the `image_x` to be the correct size for the standard case (and implying those are only related fields), but they could also get their image from a placeholder, and without having the sizes here, the placeholder will not be resized accordingly. 

Example (probably need to log on that runbot to see it): https://12224293-master-all.runbot51.odoo.com/web/image?model=res.partner&field=avatar_128&id=44
That image is not 128px max, but the original size of the placeholder (which is 180px).

See `_compute_avatar` and in particular `_avatar_get_placeholder_path`, which all produce a differently sized image than the one expected by the field, and why the field was doing (and enforcing) the size.

Reply:
I changed the condition to `self.readonly and not self.max_width and not self.max_height` to avoid that.

Otherwise this is super bloated: this controller: /web/image/user/17/avatar_128 does 
   - 3 base64.decode and 3 base64.encode (6 in total)
   - two times instanciating a PIL.Image to resize
 So, it's very slow: just to read. (normally, you should not process stuff when you read). So it's better to send a 180px placeholder and let the browser resize, rather than doing all these computation.
 
 (because a computed field do a write to assign it's value in the computation)
 
 We'll create another task to cleanup these avatar_XXX fields. (makes no sense to have 10 images fields on product & partner for the same purpose)